kedro/framework/cli/cli.py

```diff
@@ -199,6 +202,13 @@ def main(
                 click.echo(message)
                 click.echo(hint)
             sys.exit(exc.code)
+        except Exception as error:
+            logger.error(f"An error has occurred: {error}")
+            pass
+        finally:
+            self._cli_hook_manager.hook.after_command_run(
+                project_metadata=self._metadata, command_args=args, exit_code=0
+            )
```

Comment:
I think you'll have to combine the call to `after_command_run` under the first except condition, since when the command runs fine, this hook executes twice.